website_twitter_no_ext_links/models/twitter_feed.py

```diff
@@ -0,0 +1,147 @@
+# -*- coding: utf-8 -*-
+# © 2016 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import api, fields, models
+import base64, json, re, urllib, urllib2
+
+
+class TwitterFeed(models.Model):
+    _name = 'twitter.feed'
+
+    name = fields.Char(
+        string='Feed Name', 
+        help='Just a name to identify this feed',
+        required=True
+    )
+    twitter_api_key = fields.Char(
+        string='Twitter API Key',
+	help="Twitter API key you can get it from https://apps.twitter.com/app/new",
+        required=True
+    )
+    twitter_api_secret = fields.Char(
+	string='Twitter API secret',
+	help="Twitter API secret you can get it from https://apps.twitter.com/app/new",
+        required=True
+    )
+    twitter_screen_name = fields.Char(
+	string='Get favorites from this screen name',
+	help="Screen Name of the Twitter Account from which you want to load favorites."
+	"It does not have to match the API Key/Secret.",
+        required=True
+    )
+
+    feed_length = fields.Integer(string='Number of tweets to show on the website')
+
+    base_twitter_api_path = fields.Char(
+        string = 'Base twitter API path' ,
+        default="https://api.twitter.com/1.1/"
+    )
+
+    theme = fields.Many2one(
+        string="Default Theme for feed",
+        comodel_name='twitter.theme',
+        help="Default Theme, if the user does not select it from the widget it is the fallback for this feed",
+        required=True
+    )
+                    
+
+    def get_token(self, c_key, c_secret):
+
+        url = 'https://api.twitter.com/oauth2/token'
+        credentials = base64.b64encode(
+                '%s:%s' % (
+                    urllib.quote(c_key),
+                    urllib.quote(c_secret),
+                ))
+        request = urllib2.Request(url, data='grant_type=client_credentials')
+        request.add_header("Authorization", "Basic %s" % credentials)
+        request.add_header(
+            "Content-Type",
+            "application/x-www-form-urlencoded;charset=UTF-8")
+        response = urllib2.urlopen(request)
+        response_str = response.read()
+        response_dict = json.loads(response_str)
+        return str(response_dict['access_token'])
+
+    @api.model
+    def get_all_twitter_feeds(self):
+        feeds = self.env['twitter.feed'].search([])
+        for feed in feeds:
```

Comment:
put the body of this loop in its own function, which is `@api.model` but calls `self.ensure_one()` for better modularity